main.go

```diff
@@ -76,10 +90,12 @@ func main() {
 		log.Printf("Exit requested.")
 	}
 
-	log.Printf("Cleaning up pod...")
-	err = kubeClient.CoreV1().Pods(namespace).Delete(pod.Name, nil)
-	if err != nil {
-		log.Fatalf("Error cleaning up pod: %s", err)
+	if cu, err := strconv.ParseBool(cleanUpPod); cu && err != nil {
+		log.Printf("Cleaning up pod...")
+		err = kubeClient.CoreV1().Pods(namespace).Delete(pod.Name, nil)
+		if err != nil {
```

Comment:
You probably want to ignore `IsNotFoundError`

Reply:
Thanks for that. I'll change the logic to handle it.